sdk/monitor/opentelemetry-exporter/src/export/exporter.ts

```diff
@@ -0,0 +1,150 @@
+import { Logger } from "@opentelemetry/api";
+import { ConsoleLogger, LogLevel, ExportResult } from "@opentelemetry/core";
+import { Envelope } from "../Declarations/Contracts";
+import { ConnectionStringParser } from "../utils/connectionStringParser";
+import { HttpSender, FileSystemPersist } from "../platform";
+import { DEFAULT_EXPORTER_CONFIG, AzureExporterConfig } from "../config";
+import { BaseExporter, TelemetryProcessor, PersistentStorage, Sender } from "../types";
+import { isRetriable, BreezeResponse } from "../utils/breezeUtils";
+import { ENV_CONNECTION_STRING, ENV_INSTRUMENTATION_KEY } from "../Declarations/Constants";
+
+export abstract class AzureMonitorBaseExporter implements BaseExporter {
```

Comment:
is this `abstract` mostly for testability? Unless I missed something I only saw public class extend it. It feels like there should be a simpler way

Reply:
For now, there is no point to it being abstract. Eventually, there could be a metrics or logs exporter which would want to extend this. I'm open to simpler options here if any come to mind.

Currently the base exporter is taking care of all the behaviors trace/metric/log exporters would share: sending converted `Envelope[]`, handling filestorage persistence, telemetry processors, etc. This way, the trace exporter only needs to know how to convert `Span` --> `Envelope`.